Source/panels/mainpanel.cpp

```diff
@@ -56,72 +57,85 @@ void RenderMainButton(int buttonId, string_view text, int frame)
 		width = std::min<int>(GetLineWidth(text, GameFont12, spacing), (*PanelButton)[0].width());
 	}
 	RenderClxSprite(out.subregion(position.x + ((*PanelButton)[0].width() - width) / 2, position.y + 9, width, out.h() - position.y - 9), (*PanelButtonDownGrime)[frame], { 0, 0 });
-	DrawButtonText(out, text, { position + Displacement { 0, 2 }, { PanelButtonDown.w(), 0 } }, UiFlags::ColorButtonpushed, spacing);
-}
-
-void DrawTalkButton(int buttonId)
-{
-	string_view text = _("voice");
-	Point position { 176, GetMainPanel().size.height + 101 + 18 * buttonId };
-	DrawArt(*pBtmBuff, position, &TalkButton);
-	int width = std::min<int>(GetLineWidth(text, GameFont12, 1), (*PanelButton)[0].width());
-	RenderClxSprite(pBtmBuff->subregion(position.x + (TalkButton.w() - width) / 2, position.y + 6, width, 9), (*PanelButtonGrime)[1], { 0, 0 });
-	DrawButtonText(*pBtmBuff, text, { position, { TalkButton.w(), 0 } }, UiFlags::ColorButtonface);
+	DrawButtonText(out, text, { position + Displacement { 0, 2 }, { out.w(), 0 } }, UiFlags::ColorButtonpushed, spacing);
 }
 
 } // namespace
 
 void LoadMainPanel()
 {
-	LoadArt("data\\panel8bucp.pcx", &PanelButtonDown, 6);
-	PanelButton = LoadClx("data\\panel8buc.clx");
-	PanelButtonGrime = LoadClx("data\\dirtybuc.clx");
-	PanelButtonDownGrime = LoadClx("data\\dirtybucp.clx");
-
-	// Load palette to render targets
-	UpdatePalette(&PanelButtonDown);
-	if (SDLC_SetSurfaceColors(pBtmBuff->surface, PalSurface->format->palette) <= -1)
-		ErrSdl();
-
-	RenderMainButton(0, _("char"), 0);
-	RenderMainButton(1, _("quests"), 1);
-	RenderMainButton(2, _("map"), 1);
-	RenderMainButton(3, _("menu"), 0);
-	RenderMainButton(4, _("inv"), 1);
-	RenderMainButton(5, _("spells"), 0);
+	std::optional<OwnedSurface> out;
+	constexpr uint16_t NumButtonSprites = 6;
+	{
+		OptionalOwnedClxSpriteList background = LoadOptionalClx("data\\panel8bucp.clx");
+		if (!background)
+			app_fatal(_("Please update devilutionx.mpq to the latest version"));

```

Comment:
Since this is a mandatory component the other check seams redundant?

Reply:
Done